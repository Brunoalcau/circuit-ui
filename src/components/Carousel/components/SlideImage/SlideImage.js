import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';

import Image from '../../../Image';
// import AspectRatio from '../../../AspectRatio';

import { ASPECT_RATIO } from '../../constants';

const backgroundStyles = ({ theme }) => css`
  label: carousel__slideimage;
  background: ${theme.colors.n100};
`;
// fix AspectRatio when upgrade to latest @emotion dependency
const StyledAspectRatio = styled('div')(backgroundStyles);

const imageStyles = css`
  label: carousel__image;
  img {
    object-fit: cover;
  }
`;
const StyledImage = styled(Image)(imageStyles);

const SlideImage = ({ src, alt, aspectRatio, ...props }) => (
  <StyledAspectRatio aspectRatio={aspectRatio}>
    <StyledImage src={src} alt={alt} {...props} />
  </StyledAspectRatio>
);

SlideImage.propTypes = {
  /**
   * Specifies the source URL of an image
   */
  src: PropTypes.string.isRequired,
  /**
   * Provides alternative information if a user cannot view the image,
   * e.g. because of slow connection, an error in the src attribute, or if the
   * user uses a screen reader.
   */
  alt: PropTypes.string.isRequired,
  /**
   * Image aspect ratio
   */
  aspectRatio: PropTypes.number
};

SlideImage.defaultProps = {
  aspectRatio: ASPECT_RATIO
};

export default SlideImage;
