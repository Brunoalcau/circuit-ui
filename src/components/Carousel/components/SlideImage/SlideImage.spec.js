import React from 'react';

import SlideImage from './SlideImage';

const image = {
  src: 'https://placedog.net/800/500',
  alt: 'random pic'
};

describe('SlideImage', () => {
  describe('styles', () => {
    it('should render with default styles', () => {
      const actual = create(<SlideImage src={image.src} alt={image.alt} />);

      expect(actual).toMatchSnapshot();
    });

    it('should render with custom aspect ratio', () => {
      const actual = create(
        <SlideImage src={image.src} alt={image.alt} aspectRatio={2} />
      );

      expect(actual).toMatchSnapshot();
    });
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(
        <SlideImage src={image.src} alt={image.alt} />
      );
      const actual = await axe(wrapper);

      expect(actual).toHaveNoViolations();
    });
  });
});
