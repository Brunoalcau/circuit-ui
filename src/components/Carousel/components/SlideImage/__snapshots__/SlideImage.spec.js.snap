// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideImage styles should render with custom aspect ratio 1`] = `
.circuit-2 {
  background: #FAFBFC;
}

.circuit-0 {
  display: block;
  height: auto;
  max-height: 100%;
  width: 100%;
}

.circuit-0 img {
  object-fit: cover;
}

<div
  className="circuit-2 circuit-3"
>
  <img
    alt="random pic"
    className="circuit-0 circuit-1"
    src="https://placedog.net/800/500"
  />
</div>
`;

exports[`SlideImage styles should render with default styles 1`] = `
.circuit-2 {
  background: #FAFBFC;
}

.circuit-0 {
  display: block;
  height: auto;
  max-height: 100%;
  width: 100%;
}

.circuit-0 img {
  object-fit: cover;
}

<div
  className="circuit-2 circuit-3"
>
  <img
    alt="random pic"
    className="circuit-0 circuit-1"
    src="https://placedog.net/800/500"
  />
</div>
`;
